src/stdlib/Json.ts

```diff
@@ -0,0 +1,127 @@
+import { AssociativeArray } from "../brsTypes/components/AssociativeArray";
+import { BrsArray } from "../brsTypes/components/BrsArray";
+import { Interpreter } from "../interpreter";
+import { Literal } from "../parser/Expression";
+import { randomBytes } from "crypto";
+
+import {
+    BrsBoolean,
+    BrsInvalid,
+    BrsString,
+    BrsType,
+    Callable,
+    Float,
+    Int32,
+    Int64,
+    ValueKind
+} from "../brsTypes";
+
+function brsValueOf(x: any): any {
+    if (x === null) { return BrsInvalid.Instance; }
+    let t: string = typeof x;
+    switch (t) {
+    case "boolean":
+        return BrsBoolean.from(x);
+    case "string":
+        return new BrsString(x);
+    case "number":
+        if (Number.isInteger(x)) {
+            return x >= -2_147_483_648 && x <= 2_147_483_647 ? new Int32(x) : new Int64(x);
+        }
+        return new Float(x);
+    case "object":
+        if (Array.isArray(x)) {
+            return new BrsArray(x.map(brsValueOf));
+        }
+        return new AssociativeArray(
+            Object.getOwnPropertyNames(x).map((k: string) => {
+                return {
+                    name: brsValueOf(k),
+                    value: brsValueOf(x[k])
+                };
+            })
+        );
+    default:
+        throw new Error(`brsValueOf not implemented for: ${x} <${t}>`);
+    }
+}
+
+function jsonOf(interpreter: Interpreter, x: BrsType, uid: BrsString): string {
+    switch (x.kind) {
+    case ValueKind.Invalid:
+        return "null";
+    case ValueKind.String:
+        return `"${x.toString()}"`;
+    case ValueKind.Boolean:
+    case ValueKind.Double:
+    case ValueKind.Float:
+    case ValueKind.Int32:
+    case ValueKind.Int64:
+        return x.toString();
+    case ValueKind.Object:
+        if (x instanceof AssociativeArray) {
+            try {
+                if (x.get(uid) !== BrsInvalid.Instance) {
+                    throw new Error("Nested object reference");
+                }
+                let elements = x.getElements();
+                x.set(uid, BrsBoolean.True);
+                return `{${elements.map((k: BrsString) => {
+                    return `"${k.toString()}":${jsonOf(interpreter, x.get(k), uid)}`;
+                }).join(",")}}`;
+            } finally {
+                let m: Callable | undefined = x.getMethod("delete");
+                if (m) { m.call(interpreter, uid); }
+            }
+        }
+        if (x instanceof BrsArray) {
+            return `[${x.getElements().map((el: BrsType) => { return jsonOf(interpreter, el, uid); }).join(",")}]`;
+        }
+        break;
+    case ValueKind.Callable:
+    case ValueKind.Uninitialized:
+        break;
+    default:
+        const _: never = x;
```

Comment:
If you get a minute, please let me know what this does!  Without this line I don't get any compiler errors for `x` being unused, because it's used in other branches.  But as far as I can tell it's unnecessary?

Reply:
This is the exhaustive check that you turned me on to: https://basarat.gitbooks.io/typescript/content/docs/types/discriminated-unions.html. If the switch statement above does not fully cover all types in the union, this statement will cause a compile error. To demonstrate, comment out line 82 and build. You'll see a compile error:
```
src/stdlib/Json.ts:85:15 - error TS2322: Type 'Uninitialized' is not assignable to type 'never'.
```

Reply:
Ah, I see.  It's been a while since I've read through that page 😅